src/components/compositions/notifications/index.ts

```diff
@@ -0,0 +1,154 @@
+import { computed, ref, watch } from "@vue/composition-api";
+import { web3 } from "@/api/web3";
+
+const maxLimit = 100;
+const defaultShowSeconds = 7;
+
+export enum ENotificationStatus {
+  success = "success",
+  info = "info",
+  warning = "warning",
+  error = "error",
+  pending = "pending"
+}
+
+export interface INotificationView {
+  id: number;
+  status: ENotificationStatus;
+  title: string;
+  description: string;
+  txHash?: string;
+  timestamp: Date;
+  showSeconds?: number;
+  alertOnly?: boolean;
+}
+export interface IAddNotification {
+  title: string;
+  description: string;
+  status?: ENotificationStatus;
+  txHash?: string;
+  showSeconds?: number;
+  alertOnly?: boolean;
+}
+
+export const history = ref<INotificationView[]>([]);
+
+export const alertQueue = computed(() =>
+  history.value.filter(n => n.showSeconds).reverse()
+);
+
+export const pendingQueue = computed(() =>
+  history.value.filter(n => n.status === ENotificationStatus.pending)
+);
+
+export const addNotification = (payload: IAddNotification): void => {
+  const {
+    title,
+    description,
+    status,
+    txHash,
+    showSeconds,
+    alertOnly
+  } = payload;
+
+  const notification: INotificationView = {
+    id: history.value.length,
+    status: status ?? ENotificationStatus.info,
+    title,
+    description,
+    txHash,
+    showSeconds: showSeconds ?? defaultShowSeconds,
+    alertOnly,
+    timestamp: new Date()
+  };
+
+  if (txHash) notification.status = ENotificationStatus.pending;
+
+  history.value.unshift(notification);
+
+  if (history.value.length > maxLimit) history.value.pop();
+};
+
+export const removeNotification = (id: number): void => {
+  const index = history.value
+    .map(n => {
+      return n.id;
+    })
+    .indexOf(id);
+
+  if (index >= 0) history.value.splice(index, 1);
+  else console.error("Error: Remove Notification - ID not found.");
+};
+
+export const hideAlert = (id: number): void => {
+  const index = history.value
+    .map(n => {
+      return n.id;
+    })
+    .indexOf(id);
```

Comment:
`indexOf` is faster than find `findIndex` isn't it? 

FindIndex is useful if you want to find the first element that matches to a given predicate.

But since we are searching for a specific element and have a primitive typed array given by the .map I think indexOf is the correct choice here.

Happy to change it if you can help me understand the upside of findIndex for this usecase @Velua